### PR TITLE
feat: Add tag feature

### PR DESCRIPTION
**Pull request by softwareVirus**
## Features Added
- **Tag Functionality:**
  - Added tag lists and a link to access the profile of the author.
  - Implemented tag lists and an input field to append new tags.
  - Created a tag field in the post model.
  - Implemented a tag resource to manage tags associated with posts.

## Changes Made
- Modified the `SfCreatePost.vue` component to include tag functionality.
- Added a new component `SfEditor.vue` to handle tag input and display.
- Enhanced the `SfDisplayPost.vue` component to show tags associated with posts.
- Updated the `SfPostCard.vue` component to display tags for each post.
- Modified the `Post` model in the backend to support tag functionality.
- Added a new `Tag` model and corresponding resource to manage tags.

## Other Changes
- Improved styling and layout for tag-related components and elements.
- Implemented validation checks for tag input.
- Updated the documentation and added relevant comments.

### CODE REVIEW

File: server/app/resources/tag.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            if user_id is None:
+                # If no user ID provided, retrieve all tags
+                tags = Tag.objects()
+                return [tag.to_dict() for tag in tags]
+
+            user = User.objects(id=user_id).get()
+            tags = user.tags
+            return [tag.to_dict() for tag in tags], 200
+        except Exception as e:
+            print(e)
```
Comment (aahmetdurmaz) [inline server/app/resources/tag.py]: Forgotten debugger or a deliberately left print?

File: server/app/resources/tag.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            # Save the new tag to the database
+            new_tag.save()
+
+            # Convert the new tag to a dictionary
+            tag_dict = new_tag.to_dict()
+
+            # Return the newly created tag and HTTP status code
+            return tag_dict, 201
+        except Exception as e:
+            print(e)
```
Comment (aahmetdurmaz) [inline server/app/resources/tag.py]: Forgotten debugger or a deliberately left print?

File: client/src/components/Post/CreatePost/SfEditor.vue [hunk 3]
```diff
@@ -51,19 +83,37 @@ export default {
             }
         }
     },
+    watch: {
+        selectedTags(a, s) {
+            console.log(a, s)
```
Comment (aahmetdurmaz) [inline client/src/components/Post/CreatePost/SfEditor.vue]: Forgotten debugger

File: client/src/components/Post/SfDisplayPost.vue [hunk 4]
```diff
@@ -77,6 +94,7 @@ export default {
         }
     },
     mounted() {
+        console.log(this.tags)
```
Comment (aahmetdurmaz) [inline client/src/components/Post/SfDisplayPost.vue]: Forgotten debugger

File: server/app/resources/tag.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            user = User.objects(id=user_id).get()
+            tags = user.tags
+            return [tag.to_dict() for tag in tags], 200
+        except Exception as e:
+            print(e)
+            return {"error": f"An unexpected error occurred: {str(e)}"}, 400
+
+    @jwt_required(fresh=True)
+    def post(self):
+        """
```
Comment (aahmetdurmaz) [inline server/app/resources/tag.py:57]: Is there a duplicate documentation here?

**Comment by dilekyalcin** [comment]
you should remove console.log which is not necessary, otherwise LGTM

File: server/app/resources/tag.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+        try:
+            if user_id is None:
+                # If no user ID provided, retrieve all tags
+                tags = Tag.objects()
+                return [tag.to_dict() for tag in tags]
+
+            user = User.objects(id=user_id).get()
+            tags = user.tags
```
Comment (gecicidegisken) [inline server/app/resources/tag.py:50]: I couldn't catch a field "tags" on users model. How does this work? Or is it a future feature?

File: server/app/resources/tag.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            return [tag.to_dict() for tag in tags], 200
+        except Exception as e:
+            print(e)
+            return {"error": f"An unexpected error occurred: {str(e)}"}, 400
+
+    @jwt_required(fresh=True)
+    def post(self):
+        """
+        Create a new post.
```
Comment (gecicidegisken) [inline server/app/resources/tag.py]: I guess docstring here is outdated. Even though it's clear "post" is actually "tag",  it confused me for a couple of seconds. Maybe we should consider fixing.

**Comment by hilaldedek** [comment]
I agree with the comments above. Other than that, it LGTM.

**Pull request merged**